packages/next/src/client/legacy/image.tsx

```diff
@@ -1054,27 +1043,6 @@ export default function Image({
         ) : null}
         <ImageElement {...imgElementArgs} />
       </span>
-      {priority ? (
-        // Note how we omit the `href` attribute, as it would only be relevant
-        // for browsers that do not support `imagesrcset`, and in those cases
-        // it would likely cause the incorrect image to be preloaded.
-        //
-        // https://html.spec.whatwg.org/multipage/semantics.html#attr-link-imagesrcset
-        <Head>
-          <link
-            key={
-              '__nimg-' +
-              imgAttributes.src +
-              imgAttributes.srcSet +
-              imgAttributes.sizes
-            }
-            rel="preload"
```

Comment:
Preload is now handled by React (see tests in test/integration/next-image-legacy/default/test/).

However, React will take the attributes of the first `<img />` whereas `next/legacy/image` took the attributes of the last `<img />` with the same key. For example, `<img crossOrigin="use-credentials" href="foo.png" /><img crossOrigin="" href="foo.png" />` used to result in the image being preloaded with cross-origin set to anonymous. Now it would use `use-credentials` since that was specified in the first image.

@styfle let me know if this is ok

Reply:
> Preload is now handled by React

I don't understand - how is it handled by React? Is there a `ReactDOM.preload()` call somewhere that I'm missing?

Reply:
@eps1lon React has a passive preload for images to serve as a good default but I think we still need to support the explicit preload that this library was previously rendering as a tag.

If we set the minimum react version to 19 for pages router we can just start using ReactDOM.preload() but if we are going to land this but still have a minimum of 18 we'll need to keep around this removed code for now

Reply:
You can probably reuse this function:

https://github.com/vercel/next.js/blob/68d5a3892b4edf1b62e3ece4ada324065a96b6f3/packages/next/src/client/image-component.tsx#L313

Reply:
All existing tests are passing so I need some tests that aren't covered by React's default behavior that `next/legacy/image` was handling.

Reply:
Thinking about this more, I think we'll need to this change to React 19 as a minimum and switch to using float for the legacy implementation too since the rendered preload tag won't be deduplicated from the preload we generate for the img itself

Reply:
> I think we'll need to this change to React 19 as a minimum 

That was my plan. I don't think we can practically support 18 and 19 in Pages router.

Reply:
> I think we'll need to this change to React 19 as a minimum 

That was my plan. I don't think we can practically support 18 and 19 in Pages router.